pkg/apiserver/experiment/experiment.go

```diff
@@ -143,6 +145,36 @@ func (s *Service) createExperiment(c *gin.Context) {
 		return
 	}
 
+	ctx := context.TODO()
+	affectedPods, err := selector.SelectPods(ctx, kubeCli, nil, exp.Scope.ParseSelector(), s.conf.ClusterScoped, s.conf.TargetNamespace, s.conf.AllowedNamespaces, s.conf.IgnoredNamespaces)
+	if err != nil {
+		c.Status(http.StatusInternalServerError)
+		_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(err))
+		return
+	}
+	if len(affectedPods) == 0 {
+		c.Status(http.StatusBadRequest)
+		_ = c.Error(utils.ErrInvalidRequest.New("No Pods are matching selected chaos experiment criteria"))
+		return
+	}
+	affectedEnvironments := make(map[string][]v1.Pod)
+	for _, pod := range affectedPods {
+		env, found := pod.Labels["environment"]
+		if !found || env !="customenvs"  {
+			c.Status(http.StatusBadRequest)
+			_ = c.Error(utils.ErrInvalidRequest.New(fmt.Sprintf("It's only allowed to run experiments in custom environments. Targetting '%s' env. is not allowed", env)))
+			return
+		}
+
+		envName := pod.Labels["custom-environment-name"]
```

Comment:
It is possible to implement this in a way which does not require hardcoding TW specific labels. Something along these lines: ```
restrictExperimentsToSingleTenant: true,
tenancyLabel: "custom-environment-name",
requireLabels: "environment=customenvs",
```
etc

Then later it can be extended to match principal attribute(s) with label values